File: www/js/communicationcontroller.js
// var sid = 'oSS2fvq4PD1LsV4F';
const url_base = "https://ewserver.di.unimi.it/mobicomp/accordo/";


var register = $.ajax({
    method: 'post',
    url : url_base +  "register.php",
    dataType : 'json'
      


});
  
var sid = localStorage.getItem('sid');
function getWall(sid) {
    getWallVar =  $.ajax({
        method: 'post',
        url: url_base + "getWall.php",
        data: JSON.stringify({ sid: sid }),
        dataType: 'json'
    });
    return getWallVar;
}


function getProfile() {
   getProfileVar =  $.ajax({
        method: 'post',
        url: url_base + "getProfile.php",
        data: JSON.stringify({ sid: sid }),
        dataType: 'json'
    });
    return getProfileVar;
}

function sponsors() {
    getSponsors =  $.ajax({
         method: 'post',
         url: url_base + "sponsors.php",
         data: JSON.stringify({ sid: sid }),
         dataType: 'json'
     });
     return getSponsors;
 }

 
function pay(pid1, carta) {
    if (carta.mese == 1 && carta.anno ==2021 && carta.codice == 111) {
        cc1 = {name:carta.nome,surname:carta.cognome,number:carta.numero,month:carta.mese,year:carta.anno, secret:carta.codice}
    }
    
    console.log(cc1);
    payCar = $.ajax({
        method: 'post',
        url: url_base + "pay.php",
        data: JSON.stringify({ sid: sid, pid:pid1, cc:cc1,amount:1 }),
        dataType: 'json',
        
    });
    return payVar;
}


function getChannel(ctitle) {
    getChannelVar = $.ajax({
        method: 'post',
        url: url_base + "getChannel2.php",
        data: JSON.stringify({ sid: sid, ctitle:ctitle }),
        dataType: 'json',
        
    });
    return getChannelVar;
}

function addChannel(ctitle) {
    addChannelVar = $.ajax({
        method: 'post',
        url: url_base + "addChannel.php",
        data: JSON.stringify({ sid: sid, ctitle:ctitle }),
        dataType: 'json',
        
    });
    return addChannelVar;
}

function setNameProfile(name) {
    setNameProfileVar = $.ajax({
        method: 'post',
        url: url_base + "setProfile.php",
        data: JSON.stringify({ sid: sid, name:name }),
        dataType: 'json',
        
    });
    return setNameProfileVar;
}

function setPictureProfile(picture) {
    setPictureProfileVar = $.ajax({
        method: 'post',
        url: url_base + "setProfile.php",
        data: JSON.stringify({ sid: sid, picture:picture }),
        dataType: 'json',
        
    });
    return setPictureProfileVar;
}

function getPostImage(pid) {
    getPostImageVar = $.ajax({
        method: 'post',
        url: url_base + "getPostImage.php",
        data: JSON.stringify({ sid: sid, pid:pid }),
        dataType: 'json',
        
    });
    return getPostImageVar;
}

function addTextPost(ctitle, content) {
    addTextPostVar = $.ajax({
        method: 'post',
        url: url_base + "addPost.php",
        data: JSON.stringify({ sid: sid, ctitle:ctitle, type:"t", content:content }),
        dataType: 'json',
        
    });
    return addTextPostVar;
}

function addImagePost(ctitle, content) {
    addImagePostVar = $.ajax({
        method: 'post',
        url: url_base + "addPost.php",
        data: JSON.stringify({ sid: sid, ctitle:ctitle, type:"i", content:content }),
        dataType: 'json',
        
    });
    return addImagePostVar;
}

function addLocationPost(ctitle, lat, lon) {
    addLocationPostVar = $.ajax({
        method: 'post',
        url: url_base + "addPost.php",
        data: JSON.stringify({ sid: sid, ctitle:ctitle, type:"l", lat:lat, lon:lon }),
        dataType: 'json',
        
    });
    return addLocationPostVar;
}

function getUserPicture(uid) {
    getUserPictureVar = $.ajax({
        method: 'post',
        url: url_base + "getUserPicture.php",
        data: JSON.stringify({ sid: sid, uid:uid }),
        dataType: 'json',
        
    });
    return getUserPictureVar;
}

File: main/java/com/example/accordo/CommunicationController.java
package com.example.accordo;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class CommunicationController {
    private static final String BASE_URL = "https://ewserver.di.unimi.it/mobicomp/accordo/";
    private final RequestQueue requestQueue;
    private final String user_sid;

    public CommunicationController(Context context){
        requestQueue = Volley.newRequestQueue(context);
        SharedPreferences sharedPreferences = context.getSharedPreferences("WallActivity",Context.MODE_PRIVATE);
        user_sid = sharedPreferences.getString("user_sid",null);
    }

    public void register(Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener){
        final String service_url = "register.php";
        final String url = BASE_URL + service_url;
        final JSONObject jsonBody = new JSONObject();

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request" + service_url);
        requestQueue.add(request);
    }

    public void getWall(String user_sid, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "getWall.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url + user_sid);
        requestQueue.add(request);
    }

    public void getProfile(Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "getProfile.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);
        requestQueue.add(request);
    }

    public void setProfile(String paramToSet, String paramValue, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "setProfile.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put(paramToSet, paramValue);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url + ", setting "+ paramToSet);
        requestQueue.add(request);
    }

    public void getChannel(String ctitle, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "getChannel.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("ctitle" , ctitle);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);
        requestQueue.add(request);
    }

    public void addChannel(String ctitle, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "addChannel.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("ctitle" , ctitle);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);

        requestQueue.add(request);
    }

    public void addPost(String ctitle, String type, String lat, String lon, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "addPost.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("ctitle" , ctitle);
        params.put("type" , type);
        params.put("lat" , lat);
        params.put("lon" , lon);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);

        requestQueue.add(request);
    }

    public void addPost(String ctitle, String type, String content, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "addPost.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("ctitle" , ctitle);
        params.put("type" , type);
        params.put("content" , content);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);

        requestQueue.add(request);
    }

    public void getPostImage(String pid, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "getPostImage.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("pid" , pid);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);

        requestQueue.add(request);
    }

    public void getUserPicture(String uid, Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "getUserPicture.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        params.put("uid" , uid);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);

        requestQueue.add(request);
    }

    public void sponsors(Response.Listener<JSONObject> responseListener, Response.ErrorListener errorListener) {
        final String service_url = "sponsors.php";
        final String url = BASE_URL + service_url;

        Map<String, String> params = new HashMap();
        params.put("sid", user_sid);
        JSONObject jsonBody = new JSONObject(params);

        JsonObjectRequest request = new JsonObjectRequest(url, jsonBody, responseListener, errorListener);
        Log.d("Danilo Volley", "Sending Request " + service_url);
        requestQueue.add(request);
    }

}
